sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -576,6 +576,13 @@ class Analyzer(
     /** Extracts a [[Generator]] expression and any names assigned by aliases to their output. */
     private object AliasedGenerator {
       def unapply(e: Expression): Option[(Generator, Seq[String])] = e match {
+        case Alias(g: Generator, name) if g.elementTypes.size > 1 => {
```

Comment:
Yes, agree, we should throw exception if it's not default name here. But seems very difficult to tell whether it's default name or not, as in `HiveQl`, it will named each of the column anyway for `Project`.
Any simple way for this?